### PR TITLE
add the java spark build dc template to resources

### LINKED ISSUE #31: resources.yaml does not include java-build-dc
https://github.com/radanalyticsio/oshinko-s2i/blob/master/java/javabuilddc.json

without this users do not have access to the java s2i templates.

### PR DESCRIPTION
**Pull request by elmiko**
This change adds the template for the java s2i builder deployment config
template. It copies the python template and then replaces values with
the upstream java template from the s2i repository.

closes #31

### CODE REVIEW

**Comment by elmiko** [comment]
@mattf @tmckayus ptal

File: resources.yaml [hunk 1]
```diff
@@ -180,6 +180,170 @@ items:
...
+              port: 8080
+              protocol: TCP
+              targetPort: 8080
+          selector:
+            deploymentconfig: ${APPLICATION_NAME}
+    parameters:
+      - name: APPLICATION_NAME
+        displayName: Application Name
+        description: The name to use for the BuildConfig, ImageStream and DeploymentConfig components
+        from: pyspark-[a-z0-9]{4}
```
Comment (mattf) [inline resources.yaml]: s/pyspark/java/

File: resources.yaml [hunk 2]
```diff
@@ -180,6 +180,170 @@ items:
...
+        description: The name of a stored cluster configuration to use if a cluster is created, default is 'default'.
+      - name: OSHINKO_SPARK_DRIVER_CONFIG
+        description: The name of a configmap to use for the spark configuration of the driver. If this configmap is empty the default spark configuration will be used.
+      - name: OSHINKO_DEL_CLUSTER
+        description: If a cluster is created on-demand, delete the cluster when the application finishes if this option is set to 'true'
+        value: "true"
+        required: true
+      - name: APP_FILE
+        description: The name of the main JAR file. If this is not specified and there is a single JAR produced by the build, that JAR will be chosen.
+      - name: APP_EXIT
```
Comment (mattf) [inline resources.yaml]: please mirror #34 and remove this param

File: resources.yaml [hunk 3]
```diff
@@ -180,6 +180,170 @@ items:
...
+        description: Setting this value to 'false' prevents the application from being re-deployed if/when it completes
+        value: "false"
+        required: true
+      - name: APP_MAIN_CLASS
```
Comment (mattf) [inline resources.yaml]: for a java app the main class must be high priority and not guessable. it should be higher in the list (will display higher in the console) and have a displayName.

**Comment by elmiko** [comment]
@matt thanks for catching those

**Comment by mattf** [comment]
what application or example can i use to test this?

**Comment by elmiko** [comment]
it should work with the spark pi example, https://github.com/tmckayus/turbo-octo-telegram

**Comment by mattf** [comment]
$ oc create -f https://raw.githubusercontent.com/elmiko/radanalyticsio.github.io/a50d6dbf6a627357f5f139682b55fc9e01f4936e/resources.yaml
template "oshinko-java-spark-build-dc" created

"Add to Project" from the console -> search for "spark" -> select "Java Spark" -> enter https://github.com/tmckayus/turbo-octo-telegram as GIT_URI and com.mycompany.app.JavaSparkPi as APP_MAIN_CLASS and pr37 as APPLICATION_NAME -> Cannot create deployment config "pr37". DeploymentConfig in version "v1" cannot be handled as a DeploymentConfig: [pos 621]: json: expect char '"' but got char 't'.

the imagestream, buildconfig and service were successfully created.

File: resources.yaml [hunk 4]
```diff
@@ -180,6 +180,184 @@ items:
...
+                    - name: APP_ARGS
+                      value: ${APP_ARGS}
+                    - name: SPARK_OPTIONS
+                      value: ${SPARK_OPTIONS}
+                    - name: APP_MAIN_CLASS
+                      value: ${APP_MAIN_CLASS}
+                    - name: OSHINKO_DEL_CLUSTER
+                      value: ${OSHINKO_DEL_CLUSTER}
+                    - name: APP_EXIT
+                      value: true
```
Comment (mattf) [inline resources.yaml]: i think this needs to be quoted

Comment (elmiko) [inline resources.yaml]: good spot

**Comment by mattf** [comment]
lgtm

**Pull request merged**